sdk/eventhub/Azure.Messaging.EventHubs/src/EventHubsClaimPartitionException.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Azure.Messaging.EventHubs
+{
+    using System;
+
+    /// <summary>
+    ///     The exception that is thrown when a call to claim ownership of a partition fails.  Callers retry the operation.
+    /// </summary>
+    public class EventHubsClaimPartitionException : EventHubsException
```

Comment:
Have we considered doing this by adding an additional reason in the `FailureReason` enum embedded in the Event Hubs exception?   Generally speaking, we've been asked not to add custom exceptions without a strong justification around "would a developer using the client library have a need to catch this explicitly and do something special with it?"

In this case, it looks like we're intending this mainly for internal use.   I'm thinking that the failure reason combined with the [`Data`](https://docs.microsoft.com/en-us/dotnet/api/system.exception.data?view=netcore-3.1#System_Exception_Data) property to ferry over the partition identifier would be sufficient.  

Thoughts?

Reply:
An alternative would be to make this an `internal` type since the use appears to be intended for processor infrastructure.   For that, we'd need to throw it into the `Shared` project and bundle it with the processor items.

Reply:
Thanks for the context and guidance related to type best practices. I agree that tweaking `FailureReason` would be a less impactful change. I had also not considered leveraging the `Data` property. Great suggestion!

I'll incorporate that into the next commit along with some ideas around how to keep some strong typing on the extended data values.